api/shape/eachPoint.js

```diff
@@ -4,25 +4,27 @@ import Shape from './Shape.js';
 import { destructure } from './destructure.js';
 import { eachPoint as eachPointOfGeometry } from '@jsxcad/geometry';
 
-export const eachPoint = Shape.chainable((...args) => (shape) => {
-  const { shapesAndFunctions } = destructure(args);
-  let [pointOp = (point) => (shape) => point, groupOp = Group] =
-    shapesAndFunctions;
-  if (pointOp instanceof Shape) {
-    const pointShape = pointOp;
-    pointOp = (point) => (shape) => pointShape.by(point);
-  }
-  const points = [];
-  let nth = 0;
-  eachPointOfGeometry(shape.toGeometry(), ([x = 0, y = 0, z = 0]) =>
-    points.push(pointOp(Point().move(x, y, z), nth++)(shape))
-  );
-  const grouped = groupOp(...points);
-  if (grouped instanceof Function) {
-    return grouped(shape);
-  } else {
-    return grouped;
-  }
-});
-
-Shape.registerMethod('eachPoint', eachPoint);
+export const eachPoint = Shape.registerMethod(
+  'eachPoint',
+  (...args) =>
+    async (shape) => {
+      const { shapesAndFunctions } = destructure(args);
+      let [pointOp = (point) => (shape) => point, groupOp = Group] =
+        shapesAndFunctions;
+      const coordinates = [];
+      let nth = 0;
+      eachPointOfGeometry(await shape.toGeometry(), ([x = 0, y = 0, z = 0]) =>
+        coordinates.push([x, y, z])
+      );
+      const points = [];
+      for (const [x, y, z] of coordinates) {
+        points.push(await pointOp(Point().move(x, y, z), nth++))
```

Comment:
This is interesting, I haven't seen github do this kind of thing before

Reply:
Which kind of thing?

Reply:
Ah, yes -- that is nice.